app/views/unattended/provisioning_templates/snippet/kickstart_rhsm.erb

```diff
@@ -23,6 +23,9 @@ test_on:
                                                   .map { |add_on| "--addon '#{add_on.strip}'" }.join(" ")}" if host_param('syspurpose_addons')
 -%>
 <% if subman_registration -%>
+%pre
+hostnamectl set-hostname --static <%= @host.name %>
```

Comment:
Can you please comment here why it is necessary, the same way described in the https://projects.theforeman.org/issues/37098?

Reply:
Provisioning a RHEL9 system with a dedicated nic for provisioning and another as primary results in a duplicated host created on satellite, with the name defined on the provisioning interface.

This happens because we're using the rhsm component of Anaconda to register the system. At that point of the installation, the host is using the hostname that it got from the DHCP. Then it registers into Satellite with that name.

Later, the anaconda will update the name for the name of the primary interface, but at that point, the extra host already exists on Satellite.

This change sets the expected hostname to the host early during the installation avoiding this situation.